DParser2/Refactoring/TypeReferenceFinder.cs

```diff
@@ -571,9 +571,16 @@ public override void Visit (PostfixExpression_Access x)
 					var kind = TypeReferenceKind.MemberVariable;
 					if (resolveTypes)
 					{
-						var type = LooseResolution.ResolveTypeLoosely(editorData, x, out _, false);
-						if(type != null)
-							kind = type.Accept(typeTypeDet);
+						try
+						{
+							var type = LooseResolution.ResolveTypeLoosely(editorData, x, out _, false);
+							if (type != null)
+								kind = type.Accept(typeTypeDet);
+						}
+						catch (Exception)
+						{
+							// avoid cancelling everything if it fails for some, e.g. System.NotImplementedException

```

Comment:
Das geht doch nicht :///
Wenn, dann würde ich Exceptions auf API-Ebene fangen.
Welche NotImplementedExceptions waren das denn?

Reply:
Schön ist das nicht, aber die gesamte Suche abzubrechen ist auch keine Lösung ;-) Wenn ich mich recht erinnere, kam die aus `TemplateTypeParameterTypeMatcher.Visit(VarArgDecl td, ISemantic parameter)`.

Reply:
Ich denke, ich muss mir echt mal wieder Zeit nehmen, auch ein paar Sachen gerade zu ziehen :/

Reply:
Ich glaube, man muss da vielleicht anfangen, mit Futures oder Optionals zu arbeiten, um sich dieses Wirrwarr mit null-Pointern vom Hals zu schaffen^^
Und dann die ganze Suche mit allen Einzelteilen sukzessive durchzuführen.
Anstatt permanent eine ganze Liste aufzubauen, müsste man vielleicht nur soviele Ergebnisse liefern, wie auf die ersten paar Pages Completionliste passen -- dann könnte man sich in vielen Fällen diesen Wahnsinn mit UFCS etc. sparen, weil der Nutzer eh bescheid weiß, was er haben will.
Zumindest ist das meine neuere Erfahrung in IntelliJ